src/server/index.js

```diff
@@ -1,73 +1,236 @@
 'use strict'
 
-// const {waterfall} = require('async')
-const RPC = require('./rpc')
 const debug = require('debug')
-const log = debug('libp2p:rendezvous:server')
-const AsyncQueue = require('./queue')
-const BasicStore = require('./store/basic')
-
-class Server {
-  constructor (opt) {
-    if (!opt) opt = {}
-    this.node = opt.node
-    this.config = opt.config
-    this.que = new AsyncQueue()
-    this.table = {
-      NS: {},
-      RPC: {}
-    }
-    const Store = opt.store || BasicStore
-    this.store = new Store(this)
-    this._stubNS = this.store.create(Buffer.alloc(256, '0').toString())
+const log = debug('libp2p:rendezvous-server')
+log.error = debug('libp2p:rendezvous-server:error')
+
+const PeerId = require('peer-id')
+
+const { PROTOCOL_MULTICODEC, MAX_LIMIT } = require('../constants')
+const rpc = require('./rpc')
+
+/**
+* Rendezvous registration.
+* @typedef {Object} Register
+* @property {string} ns
+* @property {Buffer} signedPeerRecord
+* @property {number} ttl
+*/
+
+/**
+ * Namespace registration.
+ * @typedef {Object} NamespaceRegistration
+ * @property {string} id
+ * @property {number} expiration
+ */
+
+/**
+ * Libp2p rendezvous server.
+ */
+class RendezvousServer {
+  /**
+     * @constructor
+     * @param {Libp2p} libp2p
+     * @param {object} [options]
+     * @param {number} [options.gcInterval = 3e5]
+     */
+  constructor (libp2p, { gcInterval = 3e5 } = {}) {
```

Comment:
Why is the default 3 seconds? This seems really low, the default ttl is 2 hours.